### ISSUE TITLE
k8s: Pods should not be deployed into default namespace.

### ISSUE THREAD

**Issue by technosophos** [deis/deis#4161]
Each pod in a namespace is given information about every other pod in that namespace. This applies to the default namespace. So if I deploy two apps, each one has env vars like this:

```
APP_14334_PORT=tcp://10.100.119.52:80
APP_14334_PORT_80_TCP=tcp://10.100.119.52:80
APP_14334_PORT_80_TCP_ADDR=10.100.119.52
APP_14334_PORT_80_TCP_PORT=80
APP_14334_PORT_80_TCP_PROTO=tcp
APP_14334_SERVICE_HOST=10.100.119.52
APP_14334_SERVICE_PORT=80
APP_28146_PORT=tcp://10.100.37.242:80
APP_28146_PORT_80_TCP=tcp://10.100.37.242:80
APP_28146_PORT_80_TCP_ADDR=10.100.37.242
APP_28146_PORT_80_TCP_PORT=80
APP_28146_PORT_80_TCP_PROTO=tcp
APP_28146_SERVICE_HOST=10.100.37.242
APP_28146_SERVICE_PORT=80
DEIS_APP=kabuki-quadrant
DEIS_RELEASE=v13
GIT_SHA=3e1d7e90c8ceec6ecbc34a1e1edc31c5ff892d9e
HOME=/root
...
```

It would be better practice for the preview release if we just put every _app_ into its own namespace (so `APP_14334` is also in the namespace `APP_14334`).

This would require making hopefully simple changes to the RC and Service definitions. But it also requires explicitly creating a new namespace.

**Comment by gabrtv** [comment]
@technosophos wouldn't this prevent inter-app discovery/communication?

**Comment by technosophos** [comment]
Currently, if you deploy an app using any other scheduler in Deis, that app has env vars that pertain to itself.

The k8s preview, though, sets 7 environment variables for _every single app_ that is deployed in Deis. Every single app gets information on how to connect to every other app throughout Deis.

To make k8s work the way current Deis works, we should give each app its own namespace so that it will (by default) only know about itself.

In the future, I think we can do some fantastic things with the namespacing feature. But for the preview, I feel like we should stay as close to the current Deis scheme as we can.

**Comment by mboersma** [comment]
Fixed by #4170.
